src/hotspot/share/cds/classListParser.hpp

```diff
@@ -28,11 +28,12 @@
 #include "utilities/exceptions.hpp"
 #include "utilities/globalDefinitions.hpp"
 #include "utilities/growableArray.hpp"
-#include "utilities/hashtable.inline.hpp"
+#include "utilities/resizeableResourceHash.hpp"
 
 #define LAMBDA_PROXY_TAG "@lambda-proxy"
 #define LAMBDA_FORM_TAG  "@lambda-form-invoker"
 
+class constantPoolHandle;
 class Thread;
```

Comment:
I was looking for the use of constantPoolHandle in the header and I know why the forward declaration is needed.  Shouldn't this declaration use a const reference so that the handle code doesn't create an unnecessary copy?

bool ClassListParser::is_matching_cp_entry(constantPoolHandle &pool, int cp_index, TRAPS) {

Reply:
I changed it to pass `const constantPoolHandle &pool`. As we discussed offline, neither will create a new copy of the handle, but passing a `const` reference is cleaner.